internal/util/k8sutil/crd.go

```diff
@@ -53,10 +58,96 @@ func GetCRDManifestPaths(crdsDir string) (crdPaths []string, err error) {
 		if info == nil {
 			return nil
 		}
-		if !info.IsDir() && strings.HasSuffix(path, "_crd.yaml") {
-			crdPaths = append(crdPaths, path)
+		if !info.IsDir() {
+			b, err := ioutil.ReadFile(path)
+			if err != nil {
+				return errors.Wrapf(err, "error reading manifest %s", path)
+			}
+			typeMeta, err := GetTypeMetaFromBytes(b)
+			if err != nil {
+				return errors.Wrapf(err, "error getting kind from manifest %s", path)
+			}
+			if typeMeta.Kind == "CustomResourceDefinition" {
+				crdPaths = append(crdPaths, path)
+			}
 		}
 		return nil
 	})
 	return crdPaths, err
 }
+
+// ParseGroupSubpackages parses the apisDir directory tree and returns a map of
+// all found API groups to subpackages.
+func ParseGroupSubpackages(apisDir string) (map[string][]string, error) {
+	return parseGroupSubdirs(apisDir, false)
+}
+
+// ParseGroupSubdirs parses the apisDir directory tree and returns a map of
+// all found API groups to versions.
+func ParseGroupVersions(apisDir string) (map[string][]string, error) {
+	return parseGroupSubdirs(apisDir, true)
+}
+
+var versionRegexp = regexp.MustCompile("^v[1-9][0-9]*((alpha|beta)[1-9][0-9]*)?$")
+
+func parseGroupSubdirs(apisDir string, strict bool) (map[string][]string, error) {
```

Comment:
Perhaps add a header comment for parseGroupSubdirs as well to explain `strict` and the return values.
Not immediately obvious what that means. I had to double check to understand that it lets you filter out pkg names under at `pkg/apis/group/<pkg-name>` that don't follow the k8s API Version formatting convention.

And WDYT about renaming `strict` to something like `strictK8sVersionFormat` or `strictVersionFormat`?